src/contexts/ModalContext.tsx

```diff
@@ -76,11 +77,37 @@ export default function ModalContext(
         closeModal,
       }}
     >
-      <Dialog open={!!state} onClose={closeModal} className="relative z-50">
-        <div className="z-10 fixed inset-0 bg-black/50" aria-hidden="true" />
-        {state?.Modal /** should always be wrapped with Dialog.Panel */}
-      </Dialog>
-
+      <Transition show={!!state} as={Fragment}>
+        <Dialog onClose={closeModal} className="relative z-50">
+          <Transition.Child
+            as={Fragment}
+            enter="ease-out duration-300"
+            enterFrom="opacity-0"
+            enterTo="opacity-100"
+            leave="ease-in duration-300"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <div
+              className="z-10 fixed inset-0 bg-black/50"
+              aria-hidden="true"
+            />
+          </Transition.Child>
+          <Transition.Child
+            as={Fragment}
```

Comment:
```suggestion

```

Reply:
I will check this 
 I tried a similar approach but it didnt work cause since transition.child wasnt rendered by default on render transition effect didnt occur, 
 I will try the suggested approach

Reply:
@ap-justin tried your approach but faced similar issue, since Transition.Child is not rendered already the transition effect is not visible , I think for now we need to go with this approach

Reply:
@abbasali1994, not sure why it's not working for you - works fine with me 

the extra `div` is problematic
* in the current solution, `backdrop` seem to go in front of `modal` and transition is not smooth
https://www.loom.com/share/3d1f7a2f4b1d4b979195371c4de837d2
* when applied with other transition effects e.g. `scale` , modal breaks
https://www.loom.com/share/29374750e0df4fa0b8c6c3412fe3006e

working when `<Transition/>` is rendered along with `<Dialog.Panel/>`
* opacity transition accurate
https://www.loom.com/share/015763346e92461cb2bbe48f09a37b7c

* scale transition correct
https://www.loom.com/share/963923e149e74a1bbfe5a5ddf313b808

plus to extra `div` which is kind of out of place, since most `panels` are `fixed` positioned
```
<div>
 <modal fixed />
</div>
```

Reply:
@abbasali1994, see example #1850

Reply:
@ap-justin yes justin i figured out what i was doing out wrong yesterday and updated the PR sorry didnt push the commit, 
i hv taken reference from your PR and modified it furthur, Please check now